Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -820,6 +861,79 @@ public Task Sort( DataGridColumn<TItem> column, SortDirection? sortDirection = n
         return Task.CompletedTask;
     }
 
+
+    /// <summary>
+    /// Replaces the current sorting with the given sort order
+    /// </summary>
+    /// <param name="columns">Columns used for sorting</param>
+    /// <returns>A task that represents the asynchronous operation.</returns>
+    /// <remarks>
+    /// Note that <see cref="DataGridColumn{TItem}.Sortable"/> and <see cref="Sortable"/> must be enabled to be able to sort!
+    /// If more than one column is specified, <see cref="SortMode"/> must be <see cref="DataGridSortMode.Multiple"/>
+    /// </remarks>
+    public async Task ApplySorting(params DataGridSortColumn[] columns ) {
+        if ( columns.Length > 1 && SortMode == DataGridSortMode.Single )
+        {
+            throw new ArgumentException(
+                $"Cannot sort on multiple columns when {nameof(SortMode)} is {nameof(DataGridSortMode.Single)}." );
+        }
+
+        if ( SortMode == DataGridSortMode.Single )
+        {
+            if ( columns.Length == 1 )
+            {
+                await Sort( columns[0].Field, columns[0].SortDirection );
+            }
+            else if ( SortByColumns.Count == 1 )
+            {
+                await Sort( SortByColumns[0].Field, SortDirection.Default );
+            }
+
+            return;
+        }
+
+        // reset sorting
+        foreach ( var column in SortByColumns )
+        {
+            column.CurrentSortDirection = SortDirection.Default;
+            await column.ResetSortOrder();
+        }
+
+        SortByColumns.Clear();
+
+        // apply new sorting
+        var columnTuples = columns
+            .Select( ( x, idx ) => (
+                Column: Columns.SingleOrDefault( c => c.Field == x.Field ),
+                Direction: x.SortDirection,
+                SortOrder: idx ) )
+            .Where( x => x.Column is { Sortable: true } &&
+                         x.Direction != SortDirection.Default );
+
+        foreach ( var (column, direction, sortOrder) in columnTuples )
+        {
+            if ( SortByColumns.All( x => x.GetFieldToSort() != column.GetFieldToSort() ) )

```

Comment:
Wasn't this cleared above?
`SortByColumns.Clear();`

Reply:
I want to make sure that no duplicates end up in `SortByColumns`. New suggestion: `DistinctBy(x)` on the list passed by the user.

Fixed in e3e51e1686c3a99a22468e12fab5646c4a061d2b